gdb-server/src/architecture.rs

```diff
@@ -141,7 +141,8 @@ impl GdbTargetExt for probe_rs::Target {
         // - armv8-m.main -> Core-M33
         // - riscv:rv32   -> RISCV
 
-        let architecture = match self.core_type {
+        // TODO: what if they're not all equal?
```

Comment:
We should remove this todo and replace it with a comment. I'd assume they are always the same until we know better. If they ever differ I'd suggest that we extend this to return an array containing all the architectures.

I think I know of no MCU where this is an issue atm.

Reply:
Ah, didn't spot this one. In one of my last commits I replaced it where something does pretty much the same thing with an assert. That way, at least we won't just assume that all cores have the same architecture and actually check it

Reply:
Sounds good!

Reply:
The assert checks all have the same arch (arm vs riscv), but this piece of code cares about the actual version (armv6 vs armv7). There are chips out there with this. For example STM32WL is M4 + M0+. So the TODO still applies.

What do we do? I don't think GDB even supports debugging mixed-version chips like that. We can expose cores as "threads" to GDB, but GDB assumes they're all the same arch version.

Reply:
Ok, leave it as it is for now and revisit later. TBH I am all in favor of letting GDB die and go all in on our own DAP & debugging implementation once that hits beta/all basic features and then we do not have to care anymore.